src/dune_rpc_impl/server.ml

```diff
@@ -248,7 +248,8 @@ let handler (t : _ t Fdecl.t) action_runner_server handle : 'a Dune_rpc_server.H
                match prev, now with
                | None, None -> assert false
                | Some prev, None ->
-                 Some (Diagnostics.diagnostic_event_of_error_event (Remove prev))
+                 Some
+                   (Diagnostics.diagnostic_event_of_error_event (Remove (Error.id prev)))
```

Comment:
Small nitpick: we have the id available directly from the key passed to f, so we shouldn't need to pull it out of prev.

Reply:
That seems reasonable. Shall we do that in another PR? The behaviour here should try to stay identical.

Reply:
Wouldn't behavior remain the same? I think it'd be fine to do it in this PR, but I'm also fine with doing it in a separate one.